src/ops/dataflow.rs

```diff
@@ -115,8 +111,8 @@ impl<T: DataflowOpTrait> OpTrait for T {
     fn tag(&self) -> OpTag {
         T::TAG
     }
-    fn signature(&self) -> FunctionType {
-        DataflowOpTrait::signature(self)
+    fn signature(&self) -> Option<FunctionType> {
+        Some(DataflowOpTrait::dataflow_signature(self))
```

Comment:
This kinda makes me think the names should be the other way around. Anything defining `dataflow_signature` definitely has one, whereas the broader set of nodes defining `signature` might or might not...(but if they do, it's a dataflow signature, no?)

Reply:
As in, I suggest `DataflowOpTrait::signature(&self) -> FunctionType` and `OpTrait::dataflow_signature(&self) -> Option<FunctionType>`